### PR TITLE
[Justice Counts] Metrics Not Uploaded section in bulk upload review page

### PR DESCRIPTION
**Pull request by morden35**
## Description of the change

Previously, we were adding a 'Missing Metric' warning if no data (aggregate nor breakdown) was provided for a given message. This resulted in many messages appearing in the 'Saved Metrics' section on the Bulk Upload Review page with the same 'Missing Metric' error.

Now, we are no longer raising a 'Missing Metric' warning. Instead, I have created a separate 'Metrics Not Uploaded' section at the bottom of the Bulk Upload Review page. This section contains all metrics that had no data provided upon bulk upload.

Screenshot of entire page:
Screenshot 2023-03-28 at 5 06 21 PM

Close up of new section:
Screenshot 2023-03-28 at 5 05 58 PM

If a metric is disabled and not uploaded, it appears in the new section greyed out:
Screenshot 2023-03-29 at 10 13 27 AM

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Related to [#19774](https://github.com/Recidiviz/recidiviz-data/pull/19774)
Closes [#19687](https://app.zenhub.com/workspaces/justice-counts--622109ea3bb6e0001d2d06ad/issues/gh/recidiviz/recidiviz-data/19687)

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/DataUpload/UploadErrorsWarnings.tsx [hunk 1]
```diff
@@ -76,6 +76,8 @@ export const UploadErrorsWarnings: React.FC<UploadErrorsWarningsProps> = ({
     selectedSystem && systemToTemplateSpreadsheetFileName[selectedSystem];
   const successfulMetricsCount =
     errorsWarningsAndSuccessfulMetrics.successfulMetrics.length;
+  const noMetricUpload =
```
Comment (mxosman) [inline publisher/src/components/DataUpload/UploadErrorsWarnings.tsx:79]: nit: I would add `Count` to this var name

**Comment by mxosman** [comment]
This looks awesome, Michelle! I love that there's a section for not uploaded metrics! Thank you for the descriptive PR and screenshots - super helpful! One question I had was whether or not the disabled metrics in the "Metrics Not Uploaded" section should get the `metricDisabled` text?

**Review by lilidworkin — APPROVED** [review]
This looks SO great, @morden35 ! I really love your design choice here!

> One question I had was whether or not the disabled metrics in the "Metrics Not Uploaded" section should get the metricDisabled text?

Interesting! Now that I think about it though, not sure it's necessary. The user didn't upload this metric, so they either know it's disabled, or they don't care. I feel like it's more important to explicitly call out that its disabled and direct them to Metric Config if they actually did try to upload data for it.

If it's easy, we could show the disabled icon though, in addition to the grey text, which is somewhat easy to miss? This is the icon I mean: 
Screenshot 2023-03-29 at 8 24 55 PM

But I wouldn't spend more than like 10 minutes on this, haha. @mxosman if you have a pointer or thoughts on how we could accomplish this / if it's easy, let us know!

**Comment by mxosman** [comment]
> This looks SO great, @morden35 ! I really love your design choice here!
> 
> > One question I had was whether or not the disabled metrics in the "Metrics Not Uploaded" section should get the metricDisabled text?
> 
> Interesting! Now that I think about it though, not sure it's necessary. The user didn't upload this metric, so they either know it's disabled, or they don't care. I feel like it's more important to explicitly call out that its disabled and direct them to Metric Config if they actually did try to upload data for it.
> 
> If it's easy, we could show the disabled icon though, in addition to the grey text, which is somewhat easy to miss? This is the icon I mean: Screenshot 2023-03-29 at 8 24 55 PM
> 
> But I wouldn't spend more than like 10 minutes on this, haha. @mxosman if you have a pointer or thoughts on how we could accomplish this / if it's easy, let us know!

That makes sense why it's not necessary! I agree - though I feel like an icon might be confusing without context or a tooltip? I'm totally good with how it is now - esp. since it's not a metric a user uploaded/cares about!

**Comment by lilidworkin** [comment]
Makes sense to me!! Thanks @mxosman !

**Pull request merged**